lib/puppet/rest/errors.rb

```diff
@@ -0,0 +1,14 @@
+module Puppet::Rest
+  class ResponseError < RuntimeError
```

Comment:
Should this descend from [`Puppet::Error`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/error.rb) or integrate with [`Puppet::Network::HTTP::Error`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http/error.rb)?

Reply:
I want to keep this stuff separate from all the `Puppet::Network` stuff for now, but descending from `Puppet::Error` seems good.